armi/settings/caseSettings.py

```diff
@@ -368,13 +368,62 @@ def writeToYamlFile(self, fName, style="short"):
             the current state of settings
         """
         self.path = pathTools.armiAbsPath(fName)
+        if style == "medium":
+            originalSettings = self.getOriginalCaseSettings(self.path)
+        else:
+            originalSettings = None
         with open(self.path, "w") as stream:
-            writer = self.writeToYamlStream(stream, style)
+            writer = self.writeToYamlStream(stream, style, originalSettings)
         return writer
 
-    def writeToYamlStream(self, stream, style="short"):
-        """Write settings in yaml format to an arbitrary stream."""
-        writer = settingsIO.SettingsWriter(self, style=style)
+    def getOriginalCaseSettings(self, fPath):
+        """
+        Grabs the list of settings in the user-defined input file so that the settings
+        can be tracked outside of a Settings Object
+
+        Parameters
+        ----------
+        fPath : str
+            The absolute file path of the settings file
+
+        Returns
+        -------
+        list(originalSettings.keys()) : list
+            The keys of the settings read in from a yaml settings file
+        """
+        from ruamel.yaml import YAML
+
+        # We do not want to load these as settings, but just grab the dictionary straight
+        # from the settings file to know which settings are user-defined
+        with open(fPath, "r") as stream:
+            yaml = YAML()
+            tree = yaml.load(stream)
+            originalSettings = tree[settingsIO.Roots.CUSTOM]
+        return list(originalSettings.keys())
+
+    def writeToYamlStream(self, stream, style="short", originalSettingsNames=None):
```

Comment:
Or maybe not, if it is possible to run a case with a completely blank case settings file?

Reply:
Well even if the case settings were blank, `getOriginalCaseSettings()` would still return an empty list.

Reply:
`short` never needs to know about `originalSettingsNames` since it is only writing non-defaults. I see what you mean though, which you're right, it could end up writing a blank file! I guess that was a risk before any of this. I can add a check to the writer to abort the writing if the `settingData` is empty, if you think that would be helpful.

Reply:
Whoops, that was a typo before. I meant to say `style == "medium"`

Reply:
Oh! Yes we can't have the comparison happen if the list is None. Thanks for bringing that up.

If the (renamed) `settingsSetByUser` list defaults instead to an empty list then the comparison would happen without failure. Then it would give back an empty `settingsData` dictionary. Is that OK? Or should I add something to stop the writing?

Reply:
Yeah, sounds like a fine plan.